File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;

        for(int num : nums) {
            if(containsEvenNumDigits(num)) {
                count++;
            }
        }   

        return count;
    }

    private boolean containsEvenNumDigits(int num) {
        int count = 0;

        while(num > 0) {
            count++;
            num = num/10;
        }

        return count % 2 == 0 ? true : false;
    }
}

File: 1421-find-numbers-with-even-number-of-digits/1421-find-numbers-with-even-number-of-digits.go
func findNumbers(nums []int) int {
    count := 0

    for _, num := range nums {
        if hasEvenNumbersOfDigits(num) {
            count++;
        }
    }

    return count
}

func hasEvenNumbersOfDigits(num int) bool {
    digitCount := 0

    for num != 0 {
        num = num / 10;
        digitCount++;
    }

    return digitCount%2 == 0
}